### PR TITLE
Issue 7968: Add a hook to properly link speedreader rust library.

### PR DESCRIPTION
**Pull request by iefremov**
## Submitter Checklist:
For https://github.com/brave/brave-core/pull/4496

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] If adding a new directory with NPM packages, ensure this directory is
  added to `scripts/audit.js`.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
       },
       "brave-core": {
         "dir": "src/brave",
-        "branch": "master",
+        "branch": "ie_speedreader",
```
Comment (iefremov) [inline package.json]: will revert once CI is done

Comment (AndriusA) [reply package.json]: cc @mihaiplesa 

we shouldn't need to do this in brave-browser, I believe CI for every PR branch creates a corresponding branch on b-b pointing to the right branch on b-c?

Comment (iefremov) [reply package.json]: yes, but there is something else in this PR

File: lib/test.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ const test = (suite, buildConfig = config.defaultBuildConfig, options) => {
   } else {
     util.run('ninja', ['-C', config.outputDir, "fix_brave_test_install_name"], config.defaultOptions)
     util.run('ninja', ['-C', config.outputDir, "fix_brave_test_install_name_adblock"], config.defaultOptions)
+    util.run('ninja', ['-C', config.outputDir, "fix_brave_test_install_name_speedreader"], config.defaultOptions)
```
Comment (mihaiplesa) [inline lib/test.js]: this should have gone in https://github.com/brave/brave-browser/blob/ie_speedreader/lib/test.js

File: package.json (hunk 1 above)
Comment (mihaiplesa) [reply package.json]: CI puts the b-c branch automatically if it exists, just has to match the name of the b-b one (it won't push the change). You can just include your changes in `ie_speedreader` and not change `package.json`.

**Comment by iefremov** [comment]
thanks @mihaiplesa I didn't know that magic with same names :)

**Pull request closed without merging**